.changelog/1031.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+helper/resource: `TestStep.Taint` now calls `terraform taint` command ([#1030](https://github.com/hashicorp/terraform-plugin-sdk/issues/1030))
```

Comment:
I think this is more of a bug fix for existing functionality, rather than something new 😄 

``````suggestion
```release-note:bug
helper/resource: Fixed `TestStep` type `Taint` field usage to properly recreate resources
``````